flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/mapper/ModelMapper.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.mapper;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.ml.api.misc.param.Params;
+import org.apache.flink.ml.common.model.ModelSource;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import java.util.List;
+
+/**
+ * Abstract class for mappers with model.
+ */
+public abstract class ModelMapper extends Mapper {
+
+	/**
+	 * Specify where to load model data.
+	 */
+	private ModelSource modelSource;
+
+	/**
+	 * Field names of the model.
+	 */
+	private final String[] modelFieldNames;
+
+	/**
+	 * Field types of the model.
+	 */
+	private final DataType[] modelFieldTypes;
+
+	public ModelMapper(TableSchema modelSchema, TableSchema dataSchema, Params params, ModelSource modelSource) {
+		super(dataSchema, params);
+		this.modelFieldNames = modelSchema.getFieldNames();
+		this.modelFieldTypes = modelSchema.getFieldDataTypes();
+		this.modelSource = modelSource;
+	}
+
+	protected TableSchema getModelSchema() {
+		return TableSchema.builder().fields(this.modelFieldNames, this.modelFieldTypes).build();
+	}
+
+	/**
+	 * Load model from the list of Row type data.
+	 *
+	 * @param modelRows the list of Row type data
+	 */
+	public abstract void loadModel(List<Row> modelRows);
+
+	@Override
+	public final void open(Configuration parameters) {
+		Preconditions.checkState(this.modelSource != null, "model source not set.");
+		List<Row> rows = this.modelSource.getModelRows(getRuntimeContext());
```

Comment:
Shouldn't the parameters be part of the interface signature?

Reply:
Actually it shouldn't, we're working on an update which decouples ModelMapper and Mapper  from flink's `MapFunction`.

Reply:
We have updated the code.

Reply:
I see. however at this current stage it seemed weird that the open method does not use `Configuration` passed in, instead using the more generic `getRuntimeContext`. 
I am find with current impl actually. thanks for the explanation.